tt_metal/hw/inc/dataflow_api.h

```diff
@@ -211,24 +212,31 @@ constexpr static std::int32_t GET_TILE_SIZE(uint format) {
     };
 }
 
-FORCE_INLINE
-constexpr static std::uint32_t MUL_WITH_TILE_SIZE(uint format, uint index) {
+template <uint32_t tile_hw = 1024>
+FORCE_INLINE constexpr static std::uint32_t MUL_WITH_TILE_SIZE(uint format, uint index) {
+    constexpr uint8_t datum_shift = (tile_hw == 1024) ? 10 :
```

Comment:
May be for this and exp_shift, a templated function using constexpr if-else chain will be more scalable , without any additional penalty. 

upto you.